### ISSUE TITLE
GNN last_valid_pe buffer causes shape mismatch warning on checkpoint load

### ISSUE THREAD

**Issue by The-Obstacle-Is-The-Way** [Clarity-Digital-Twin/brain-go-brr-v2#8]
## Summary

When loading a checkpoint, a warning appears about shape mismatch for the `gnn.last_valid_pe` buffer. The buffer is saved with batch-specific shape but model initializes with placeholder shape.

## Warning Message

```
[CHECKPOINT] Skipping dynamic buffer with shape mismatch: gnn.last_valid_pe
(checkpoint: torch.Size([8, 192, 19, 16]), model: torch.Size([1, 1, 1, 16]))
```

## Root Cause

In `gnn_pyg.py:137-142`:
```python
# Model initializes with placeholder shape (1,1,1,k)
self.register_buffer(
    "last_valid_pe",
    torch.zeros(1, 1, 1, k_eigenvectors, dtype=torch.float32),
    persistent=True,
)
```

But during forward pass (`gnn_pyg.py:404`), it gets overwritten with actual batch shape:
```python
# Gets updated with (B, T, N, k) shape during forward
self.last_valid_pe = pe.detach().clone()
```

When checkpoint is saved, it has `(B=8, T=192, N=19, k=16)`.
When model is re-initialized for resume, it has `(1, 1, 1, 16)`.

## Current Handling

`checkpoint.py:178-183` already handles this gracefully:
```python
if key.endswith(".last_valid_pe"):
    logger.info(f"[CHECKPOINT] Skipping dynamic buffer with shape mismatch: {key}")
    buffers_to_skip.append(key)
```

## Impact

- Warning appears but is harmless (buffer gets recomputed on first forward pass)
- PE fallback mechanism works correctly after resume
- Just noisy console output

## Proposed Fixes

**Option A**: Don't save dynamic buffers
```python
self.register_buffer("last_valid_pe", ..., persistent=False)
```

**Option B**: Save scalar placeholder that reconstructs at load time

**Option C**: Accept current behavior (warning is informative, not harmful)

## Files

- `src/brain_brr/models/gnn_pyg.py:133-142` - Buffer registration
- `src/brain_brr/models/gnn_pyg.py:404` - Buffer update
- `src/brain_brr/train/checkpoint.py:178-183` - Shape mismatch handling

## Priority

Low - Warning only, no functional impact

## Labels

enhancement, model
